--- sapi/cgi/config9.m4
+++ sapi/cgi/config9.m4
@@ -88,10 +88,10 @@
     PHP_ADD_MAKEFILE_FRAGMENT($abs_srcdir/sapi/cgi/Makefile.frag)
     case $host_alias in
       *cygwin* )
-        SAPI_CGI_PATH=sapi/cgi/php.exe
+        SAPI_CGI_PATH=sapi/cgi/php-cgi.exe
         ;;
       * )
-        SAPI_CGI_PATH=sapi/cgi/php
+        SAPI_CGI_PATH=sapi/cgi/php-cgi
         ;;
     esac
     PHP_SUBST(SAPI_CGI_PATH)
@@ -145,7 +145,7 @@
     AC_DEFINE_UNQUOTED(PHP_FCGI_STATIC, $PHP_FCGI_STATIC, [ ])
     AC_MSG_RESULT($PHP_ENABLE_FASTCGI)
 
-    INSTALL_IT="@echo \"Installing PHP CGI into: \$(INSTALL_ROOT)\$(bindir)/\"; \$(INSTALL) -m 0755 \$(SAPI_CGI_PATH) \$(INSTALL_ROOT)\$(bindir)/\$(program_prefix)php\$(program_suffix)\$(EXEEXT)"
+    INSTALL_IT="@echo \"Installing PHP CGI into: \$(INSTALL_ROOT)\$(bindir)/\"; \$(INSTALL) -m 0755 \$(SAPI_CGI_PATH) \$(INSTALL_ROOT)\$(bindir)/\$(program_prefix)php-cgi\$(program_suffix)\$(EXEEXT)"
     PHP_SELECT_SAPI(cgi, program, $PHP_FCGI_FILES cgi_main.c getopt.c, , '$(SAPI_CGI_PATH)')
 
     case $host_alias in
