--- ext/openssl/openssl.c
+++ ext/openssl/openssl.c
@@ -203,8 +203,10 @@
 static char default_ssl_conf_filename[MAXPATHLEN];
 
 struct php_x509_request {
-	LHASH * global_config;	/* Global SSL config */
-	LHASH * req_config;		/* SSL config for this request */
+	// LHASH * global_config;	/* Global SSL config */
+    LHASH_OF(CONF_VALUE) * global_config; /* Global SSL config */
+	// LHASH * req_config;		/* SSL config for this request */
+    LHASH_OF(CONF_VALUE) * req_config; /* SSL config for this request */
 	const EVP_MD * md_alg;
 	const EVP_MD * digest;
 	char	* section_name,
@@ -364,7 +366,8 @@
 		const char * section_label,
 		const char * config_filename,
 		const char * section,
-		LHASH * config TSRMLS_DC)
+		// LHASH * config TSRMLS_DC)
+        LHASH_OF(CONF_VALUE) * config TSRMLS_DC)
 {
 	X509V3_CTX ctx;
 	
